lib/pbench/test/unit/agent/task/test_make_result_tb.py

```diff
@@ -31,20 +31,13 @@ def config_and_logger(self):
     def test_bad_result_dir(self, result_dir, caplog):
         caplog.set_level(logging.ERROR, logger=self.logger.name)
         if result_dir:
-            expected_error_message = f"Invalid result directory provided: {result_dir}"
+            expected_error_message = "Invalid Result directory provided"
         else:
             expected_error_message = "Result directory not provided"
-        try:
-            mrt = MakeResultTb(
-                result_dir, self.target_dir, "pbench", "", self.config, self.logger
-            )
+        with pytest.raises(FileNotFoundError) as e:
+            mrt = MakeResultTb(result_dir, self.target_dir, self.config, self.logger)
             mrt.make_result_tb()
-        except SystemExit:
-            assert caplog.records
-            assert len(caplog.records) == 1
-            assert caplog.records[0].message == expected_error_message
-        else:
-            assert False
+        assert f"{str(e).startswith(expected_error_message)}"
```

Comment:
Using an f-string here is weird. You're string-ifying `True` or `False` ... and I think a string of `"False"` is falsey so it may work, but it's still weird! Same in several other test cases...

Reply:
will change that

Reply:
@dbutenhof, a string of `"False"` has a value of `True` in boolean context.  (Other than `False` and `0`, I believe that only "empty things" -- strings, lists, tuples, dicts -- are falsey.)

So, this is "working" (i.e., not failing...), because neither `assert "True"` nor `assert "False"` fails.